### PR TITLE
small internal code cleanup

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Small cleanup of code that claimed some deprecated functionality could be removed in current devel.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: -
  - [ ] Backport for 3.9: -
  - [ ] Backport for 3.8: -

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Graph/Enumerators/OneSidedEnumeratorInterface.h [hunk 1]
```diff
@@ -73,23 +70,19 @@ class TraversalEnumerator {
...
+  using VertexRef = velocypack::HashedStringRef;
+  TraversalEnumerator() {}
   virtual ~TraversalEnumerator() {}
 
-  // NOTE: keepPathStore is only required for 3.8 compatibility and
-  // can be removed in the version after 3.9
   virtual void clear(bool keepPathStore) = 0;
   [[nodiscard]] virtual bool isDone() const = 0;
 
-  // NOTE: keepPathStore is only required for 3.8 compatibility and
```
Comment (jsteemann) [inline arangod/Graph/Enumerators/OneSidedEnumeratorInterface.h:91]: This code comment turned out not to be true: SmartGraphEngine.cpp can call the function with a `keepPathStore` value of `true`.

File: arangod/Graph/Enumerators/OneSidedEnumeratorInterface.h [hunk 2]
```diff
@@ -73,23 +70,19 @@ class TraversalEnumerator {
...
+      graph::OneSidedEnumeratorOptions&& enumeratorOptions, bool useTracing)
+      -> std::unique_ptr<TraversalEnumerator>;
 
-  using VertexRef = arangodb::velocypack::HashedStringRef;
-  TraversalEnumerator(){};
+  using VertexRef = velocypack::HashedStringRef;
+  TraversalEnumerator() {}
   virtual ~TraversalEnumerator() {}
 
-  // NOTE: keepPathStore is only required for 3.8 compatibility and
```
Comment (jsteemann) [inline arangod/Graph/Enumerators/OneSidedEnumeratorInterface.h:86]: This code comment turned out not to be true: SmartGraphEngine.cpp can call the `reset()` function with a `keepPathStore` value of `true`, which can then call the `clear()` function with a value of `true`.

File: arangod/Graph/PathManagement/PathValidatorOptions.h [hunk 3]
```diff
@@ -128,7 +129,6 @@ class PathValidatorOptions {
   aql::FixedVarExpressionContext& getExpressionContext();
 
   // @brief If a graph is asked for the first vertex and that is filtered
-  // it can be removed for 3.9 => nextVersion.
```
Comment (jsteemann) [inline arangod/Graph/PathManagement/PathValidatorOptions.h:131]: This code comment turned out not to be true: SmartGraphEngine.cpp can call this function.

**Review by hkernbach — APPROVED** [review]
LGTM (still there were some jenkins issues)

**Pull request merged**